Danki2/Assets/Scripts/Camera/CustomCamera.cs

```diff
@@ -11,10 +11,26 @@ public class CustomCamera : MonoBehaviour
     [SerializeField, Range(10, 80)]
     private float angle = 40;
 
+    [SerializeField, Range(0f, 1f)]
+    private float mouseFollowFactor = 0f;
+
+    [SerializeField, Range(0f, 1f)]
+    private float smoothFactor = 0f;
+
+    private Vector3 desiredPosition;
+
+    private CameraShakeManager shakeManager = new CameraShakeManager(4);
+
     private void Update()
     {
+        if (Input.GetKeyDown(KeyCode.O))
+        {
+            shakeManager.AddCameraShake(15f, 0.3f);
+        }
+
         UpdateRotation();
         FollowTarget();
+        transform.Translate(shakeManager.GetShakeVector(), Space.World);
```

Comment:
Instead of calling transform.Translate here, it might be cleaner to change CameraShakeManager to have an "Update" method instead of GetShakeVector, that takes a Transform as a parameter and does the translate itself, then here you would just call `shakeManager.Update(transform);`. At the moment it's not very clear that GetShakeVector needs to be called every frame for things to work properly. You could even make it its own component, but that seems like overkill...